nixos/modules/services/x11/dunst.nix

```diff
@@ -0,0 +1,160 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+   cfg = config.services.dunst;
+in {
+
+options.services.dunst = {
+
+  enable = mkOption {
+    type = types.bool;
+    default = false;
+    description = ''
+      Use the dunst notifications daemon.
+    '';
+  };
+
+  iconDirs = mkOption {
+    type = types.listOf types.path;
+    default = [];
+    example = literalExample ''
+      [ "''${pkgs.gnome3.adwaita-icon-theme}/share/icons/Adwaita/48x48" ]
+    '';
+    description = ''
+      Paths to icon folders.
+    '';
+  };
+
+  extraCliOptions = mkOption {
+    type = types.str;
+    default = "";
+    description = ''
+      This gets appended verbatim to the command line of dunst.
+    '';
+  };
+
+  globalConfig = mkOption {
+    type = types.nullOr (types.attrsOf types.string);
+    default = null;
```

Comment:
Why not
```
type = with types; attrsOf str;
default = { };
```
Anyway, use `types.str` because
```
# Deprecated; should not be used because it quietly concatenates
# strings, which is usually not what you want.
string = separatedString "";
```